pkg/scalers/azure_eventhub_scaler.go

```diff
@@ -0,0 +1,256 @@
+		}
+	}
+	if unprocessedEventCountInPartition < 0 {
+		unprocessedEventCountInPartition = 0
+	}
+
+	return unprocessedEventCountInPartition, nil
+}
+
+// IsActive determines if eventhub is active based on number of unprocessed events
+func (scaler *AzureEventHubScaler) IsActive(ctx context.Context) (bool, error) {
+	runtimeInfo, err := scaler.client.GetRuntimeInformation(ctx)
+	if err != nil {
+		log.Errorf("unable to get runtimeInfo for isActive: %s", err)
+	}
+
+	partitionIDs := runtimeInfo.PartitionIDs
+
+	for i := 0; i < len(partitionIDs); i++ {
+		partitionID := partitionIDs[i]
+
+		unprocessedEventCount, err := scaler.GetUnprocessedEventCountInPartition(ctx, partitionID)
+
+		if err != nil {
+			log.Errorf("unable to get unprocessedEventCount for isActive: %s", err)
+		}
+
+		if unprocessedEventCount > 0 {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (scaler *AzureEventHubScaler) GetMetricSpecForScaling() []v2beta1.MetricSpec {
+	return []v2beta1.MetricSpec{
+		{
+			External: &v2beta1.ExternalMetricSource{
+				MetricName:         thresholdMetricName,
+				TargetAverageValue: resource.NewQuantity(scaler.metadata.threshold, resource.DecimalSI),
+			},
+			Type: eventHubMetricType,
+		},
+	}
+}
+
+func (scaler *AzureEventHubScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	totalUnprocessedEventCount := int64(0)
+	runtimeInfo, err := scaler.client.GetRuntimeInformation(ctx)
+	if err != nil {
+		return []external_metrics.ExternalMetricValue{}, fmt.Errorf("unable to get runtimeInfo for metrics: %s", err)
+	}
+
+	partitionIDs := runtimeInfo.PartitionIDs
+
+	for i := 0; i < len(partitionIDs); i++ {
+		partitionID := partitionIDs[i]
+		partitionRuntimeInfo, err := scaler.client.GetPartitionInformation(ctx, partitionID)
+		if err != nil {
+			return []external_metrics.ExternalMetricValue{}, fmt.Errorf("unable to get partitionRuntimeInfo for metrics: %s", err)
+		}
+
+		lease, err := GetLeaseFromBlobStorage(ctx, partitionID, scaler.metadata.storageConnection, scaler.metadata.storageContainerName)
+		if err != nil {
+			return []external_metrics.ExternalMetricValue{}, fmt.Errorf("unable to get lease from storage: %s", err)
+		}
+
+		unprocessedEventCount := int64(0)
+
+		unprocessedEventCount, err = scaler.GetUnprocessedEventCountInPartition(ctx, partitionID)
+		if err != nil {
+			return []external_metrics.ExternalMetricValue{}, fmt.Errorf("unable to get unprocessedEventCount for metrics: %s", err)
+		}
+
+		totalUnprocessedEventCount += unprocessedEventCount
+
+		log.Debugf("Partition ID: %s, Last Enqueued Offset: %s, Checkpoint Offset: %s, Total new events in partition: %s",
+			partitionRuntimeInfo.PartitionID, partitionRuntimeInfo.LastEnqueuedOffset, lease.Checkpoint.Offset, strconv.FormatInt(unprocessedEventCount, 10))
+	}
+
+	log.Infof("Scaling for %s total unprocessed events", strconv.FormatInt(totalUnprocessedEventCount, 10))
```

Comment:
Change to debug and include information about the event hub container name so we know what this log pertains to?